extendr-api/src/robj/mod.rs

```diff
@@ -776,22 +781,26 @@ macro_rules! make_typed_slice {
                     _ => None
                 }
             }
+
+            fn error(robj: Robj) -> Self::Error {
```

Comment:
I think there is an argument that the `as_typed_slice*` functions should return a `Result` rather than an `Option`. Because then the trait becomes much simpler.